docs/reference/developer_documentation.md

```diff
@@ -52,17 +52,17 @@ The CI process for developers' local clones and the remote repository should be
 
 ### Local
 
-Local CI is done automatically when pushing with `.pre-commit-config.yaml`, which runs *static* tests that can be run manually with the following command. 
+Local CI is done automatically when pushing with `.pre-commit-config.yaml`, which runs *static* tests that can be run manually with the following command. This command will also generate fixes to formatting errors in staged files. After running you will need to stage the fixes for these files before running again.
 ```
 pre-commit run -a
```

Comment:
I think we should describe what `pre-commit run -a` and `poetry run pre-commit` do separately, or just document the preferred one to use. Looks like the former runs on all files and the latter on staged files, which could both be useful.

Reply:
I echo what @MatthewSteen is saying here, with the additional comment that I'd recommend the phrasing `The command below will also generate...` instead